AnkiDroid/src/main/java/com/ichi2/compat/CompatV26.java

```diff
@@ -183,4 +186,46 @@ public void abandonAudioFocus(AudioManager audioManager, AudioManager.OnAudioFoc
             audioManager.abandonAudioFocusRequest(audioFocusRequest);
         }
     }
+
+    /*
+     * This method uses [Files.newDirectoryStream].
+     * Hence this method, hasNext and next should be constant in time and space.
+     */
+    @Override
+    public @Nullable FileStream contentOfDirectory(File directory) {
+        DirectoryStream<Path> paths_stream;
+        try {
+            paths_stream = Files.newDirectoryStream(directory.toPath());
+        } catch (IOException e) {
+            Timber.w(e);
+            return null;
+        }
+        Iterator<Path> paths = paths_stream.iterator();
+        return new FileStream() {
+            @Override
+            public void close() throws IOException {
+                paths_stream.close();
+            }
+
+
+            @Override
+            public boolean hasNext() {
+                try {
+                    return paths.hasNext();
+                } catch (DirectoryIteratorException e) {
+                    // According to the documentation, it's the only exception it can throws.
+                    // We encapsulate `e` in a class that is available in
+                    // API < 26.
+                    throw new com.ichi2.compat.DirectoryIteratorException(e);
```

Comment:
Overall, I think removing `DirectoryIteratorException` would make the code a /lot/ simpler

```suggestion
                    throw e.getCause()
```